mitodl/ocw-hugo-themes
# feat: add file size in resource list and resource pages

# What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/1681

# Description (What does it do?)
Adds file size in resource list pages and individual resource pages. Refer to figma designs in https://github.com/mitodl/hq/issues/1681#issuecomment-1746874289

In this PR, apart from the SCSS changes, we:
* Remove the `get_resource_thumbnail_attributes.html` partial
* Use new partial `get_resource_category.html` instead (we no longer have imgs for thumbnails)
* Introduced a `file_size_formatter.html` partial for calculating file size in respective units (converted from bytes)
* Also changed the styling of resource body page in general (apart from the file sizing fixes). Previously we used `container` class (this is still used in a lot of places in OCW) but removed it for this page as per @mbilalmughal recommendation. It was less aesthetically pleasing. Instead, for resource body page we have `resource-page-container`. 
* There's only reformatting changes in `resource_list_collapsible.html`


# How can this be tested?
Test rendering for (to ensure nothing has broken for moving resource partials from course-v2 to base-theme):
www online + www offline
course online + course offline.

Steps:
First checkout to this branch.
`yarn build /to/ocw-content-rc/ocw-www or a course/ /to/ocw-hugo-projects/ocw-www or ocw-course-v2/config-offline.yaml`  For offline themes.

`yarn start www` For online www
Make sure the builds succeed ^
from dist directory do check that the theme is alright too from any index pages.

For course, we will want to:
1. `yarn start course 6.0002-fall-2016` (using this course because this has file size content)
2. Go to: http://localhost:3000/download/
3. Check the UI.
4. Click on a resource (other than video lectures)
5. Check the UI. Also you might want to use bigger resource names (like I have done in the screenshots) to ensure the UI does not break.
6. Also try different file sizes, too small, too big. Make sure the unit is okay.
7. Also check for cases where a resource might not have file size (just clear the `file_size` from the `.md` file of a resource)

For offline course theme, build a course and go to dist/index.html and then "Browse Resources". Steps will be same as above but not as thorough testing just make sure the changes have applied.

# References
Use these figma links for reference.
https://github.com/mitodl/hq/issues/1681#issuecomment-1746874289

# Screenshots (if appropriate):


https://github.com/mitodl/ocw-hugo-themes/assets/109785089/94d91928-921e-4f2a-b4e3-b6c3a7d7a8cf

<img width="854" alt="image" src="https://github.com/mitodl/ocw-hugo-themes/assets/109785089/373d471b-f0d1-4dd2-a191-17ac37a17a71">
<img width="857" alt="image" src="https://github.com/mitodl/ocw-hugo-themes/assets/109785089/5f50b85a-8ea1-4720-a7f3-89f223975cca">
<img width="867" alt="image" src="https://github.com/mitodl/ocw-hugo-themes/assets/109785089/73f3eec2-7ba2-4a09-b646-529134ccf868">
<img width="868" alt="image" src="https://github.com/mitodl/ocw-hugo-themes/assets/109785089/6feabd54-8d64-4d46-a2e3-11772aedbf65">
<img width="602" alt="image" src="https://github.com/mitodl/ocw-hugo-themes/assets/109785089/0f3d274e-683f-4556-8fff-dbf6701b5b02">
<img width="867" alt="image" src="https://github.com/mitodl/ocw-hugo-themes/assets/109785089/2568e10f-bf22-4ee2-80c4-186fe248be78">
<img width="770" alt="image" src="https://github.com/mitodl/ocw-hugo-themes/assets/109785089/9d097549-0d98-4f9a-aaab-f10f2a490c12">
<img width="711" alt="image" src="https://github.com/mitodl/ocw-hugo-themes/assets/109785089/63ecb4ba-774b-457e-9c0c-9d8d56257478">
<img width="866" alt="image" src="https://github.com/mitodl/ocw-hugo-themes/assets/109785089/42276a50-6fe0-4a92-961f-17e5bb69d997">
<img width="1038" alt="image" src="https://github.com/mitodl/ocw-hugo-themes/assets/109785089/795a488c-a8cd-4d33-8d8d-164951ea50b3">
<img width="685" alt="image" src="https://github.com/mitodl/ocw-hugo-themes/assets/109785089/08544e8f-256b-45f0-a914-d9b020ef60b4">
<img width="581" alt="image" src="https://github.com/mitodl/ocw-hugo-themes/assets/109785089/f6b8fa96-da5e-4ef3-b634-0b6d5974a92c">
